### PR TITLE
Streaming / more low-level JSON processing

### PR DESCRIPTION
**Pull request by skjolber**
A bit more low-level JSON processing.

Work in progress; for discussion while waiting for https://github.com/zalando/logbook/pull/580. Will require adjustment.

## Description
Stream JSON events wherever possible. Pass marginally more low-level types.

## Motivation and Context
Avoid reading JSON trees into memory.

## Types of changes
- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

## Checklist:
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [ ] I have added tests to cover my changes.

### CODE REVIEW

File: logbook-json/src/main/java/org/zalando/logbook/json/PrettyPrintingJsonBodyFilter.java [hunk 1]
```diff
@@ -16,16 +19,14 @@
...
-    private final ObjectWriter writer;
+    private final JsonFactory factory;
 
-    public PrettyPrintingJsonBodyFilter() {
-        this(new ObjectMapper());
+    public PrettyPrintingJsonBodyFilter(final JsonFactory factory) {
+        this.factory = factory;
     }
 
-    public PrettyPrintingJsonBodyFilter(final ObjectMapper mapper) {
```
Comment (whiskeysierra) [inline logbook-json/src/main/java/org/zalando/logbook/json/PrettyPrintingJsonBodyFilter.java]: Keeping signature compatibility should be an option here, right?

Comment (skjolber) [reply logbook-json/src/main/java/org/zalando/logbook/json/PrettyPrintingJsonBodyFilter.java]: Sure, hold on for update.

**Comment by whiskeysierra** [comment]
@skjolber I would merge that unless you have something else that you'd like to add.

**Comment by skjolber** [comment]
@whiskeysierra I've have no more changes, go ahead and merge.

**Comment by whiskeysierra** [comment]
:+1:

**Pull request merged**